sdk/mgmtcommon/ClientRuntime/ClientRuntime/Utilities/HttpRequestSanitizer.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Rest.Utilities
+{
+    /// <summary>
+    /// Sanitizer used internall by <see cref="HttpRequestMessageWrapper"/>.
+    /// </summary>
+    internal class HttpRequestSanitizer
+    {
+        public static readonly HttpRequestSanitizer Default = new HttpRequestSanitizer();
+
+        private readonly string _redactedPlaceholder = "REDACTED";
+        private readonly HashSet<string> _headersToSanitize = new HashSet<string>() { "authorization" };
+
+        /// <summary>
+        /// Sanitize value of sensitive headers in the given <paramref name="headers"/>.
+        /// </summary>
+        /// <param name="headers">A collection of headers to sanitize.</param>
+        public void SanitizerHeaders(IDictionary<string, IEnumerable<string>> headers)
+        {
+            if (headers != null)
+            {
```

Comment:
You could exit early from this loop if `headers == null` like this and reduce the indentation on the method.

```csharp
if (headers == null)
{
  return;
}
```